pulsar-broker/src/test/java/org/apache/pulsar/compaction/ServiceUnitStateCompactionTest.java

```diff
@@ -0,0 +1,927 @@
+            futures.add(producer.newMessage().key(key).value(testValue0((i + 20 + ""))).sendAsync());
+        }
+        FutureUtil.waitForAll(futures).get();
+
+        try (Consumer<ServiceUnitStateData> consumer = pulsarClient.newConsumer(schema)
+                .topic(topic)
+                .subscriptionName("sub3")
+                .readCompacted(true)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscribe()) {
+            for (int i = 0; i < 10; i++) {
+                Message<ServiceUnitStateData> received = consumer.receive();
+                assertNotNull(received);
+                assertEquals(received.getKey(), key);
+                assertEquals(received.getValue().broker(), i + 20 + "");
+                consumer.acknowledge(received);
+            }
+            Message<ServiceUnitStateData> none = consumer.receive(2, TimeUnit.SECONDS);
+            assertNull(none);
+        }
+    }
+
+    @SneakyThrows
+    @Test
+    public void testHealthCheckTopicNotCompacted() {
+        NamespaceName heartbeatNamespaceV1 =
+                NamespaceService.getHeartbeatNamespace(pulsar.getAdvertisedAddress(), pulsar.getConfiguration());
+        String topicV1 = "persistent://" + heartbeatNamespaceV1.toString() + "/healthcheck";
+        NamespaceName heartbeatNamespaceV2 =
+                NamespaceService.getHeartbeatNamespaceV2(pulsar.getAdvertisedAddress(), pulsar.getConfiguration());
+        String topicV2 = heartbeatNamespaceV2.toString() + "/healthcheck";
+        Producer<ServiceUnitStateData> producer1 = pulsarClient.newProducer(schema).topic(topicV1).create();
+        Producer<ServiceUnitStateData> producer2 = pulsarClient.newProducer(schema).topic(topicV2).create();
+        Optional<Topic> topicReferenceV1 = pulsar.getBrokerService().getTopic(topicV1, false).join();
+        Optional<Topic> topicReferenceV2 = pulsar.getBrokerService().getTopic(topicV2, false).join();
+        assertFalse(((SystemTopic) topicReferenceV1.get()).isCompactionEnabled());
+        assertFalse(((SystemTopic) topicReferenceV2.get()).isCompactionEnabled());
+        producer1.close();
+        producer2.close();
+    }
+
+    @Test(timeOut = 60000)
+    public void testCompactionWithMarker() throws Exception {
+        String namespace = "my-property/use/my-ns";
+        final TopicName dest = TopicName.get(
+                BrokerTestUtil.newUniqueName("persistent://" + namespace + "/testWriteMarker"));
+        admin.topics().createNonPartitionedTopic(dest.toString());
+        @Cleanup
+        Consumer<ServiceUnitStateData> consumer = pulsarClient.newConsumer(schema)
+                .topic(dest.toString())
+                .subscriptionName("test-compaction-sub")
+                .subscriptionType(SubscriptionType.Exclusive)
+                .readCompacted(true)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Latest)
+                .subscribe();
```

Comment:
The consumer is useless.

Reply:
In fact, this case is an unnecessary copy from CompactionTest.  Removed it.